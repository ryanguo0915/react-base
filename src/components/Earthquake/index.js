import React  from "react";

export default function Earthquake(props) {
    const id = props.match.params.slugifiedId;
    const earthquake = props.features.find(feature => feature.id === id) || {};
    const {title, mag, time, status, tsunami, type} = earthquake.properties;
  return (
    <div className="earthquake-wrapper">
        <h3>{title}</h3>
        <table>
            <tbody>
                <tr>
                    <td>Title</td>
                    <td>{title}</td>
                </tr>
                <tr>
                    <td>Magnitude</td>
                    <td>{mag}</td>
                </tr>
                <tr>
                    <td>Time</td>
                    <td>{time}</td>
                </tr>
                <tr>
                    <td>Status</td>
                    <td>{status}</td>
                </tr>            
                <tr>
                    <td>Tsunami</td>
                    <td>{tsunami}</td>
                </tr>            
                <tr>
                    <td>Type</td>
                    <td>{type}</td>
                </tr>
            </tbody>
        </table>
    </div>
  );
}