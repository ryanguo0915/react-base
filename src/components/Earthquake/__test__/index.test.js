import React from 'react';
import { shallow } from 'enzyme';
import Earthquake from '../index';
describe("MyComponent", () => {
  it("should render my component", () => {
    const props = { 
        features: [{id:'1', properties:{title:'title', mag: 'mag', time: 'time', status: 'status', tsunami: 'tsunami', type: 'type'}}], 
        match: { params: {slugifiedId: '1'}}
    };
    const wrapper = shallow(<Earthquake {...props}/>);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
  it("should have 1 matched perperty and render tables", () => {
    const props = { 
        features: [{id:'1', properties:{title:'title', mag: 'mag', time: 'time', status: 'status', tsunami: 'tsunami', type: 'type'}}], 
        match: { params: {slugifiedId: '1'}}
    };
    const wrapper = shallow(<Earthquake {...props}/>);
    expect(wrapper.getElements()).toMatchSnapshot();
    const table = wrapper.find('table');
    const row = wrapper.find('tr');
    expect(table).toHaveLength(1);
    expect(row).toHaveLength(6);
  });
});