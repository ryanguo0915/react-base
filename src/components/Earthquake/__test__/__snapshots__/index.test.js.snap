// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should have 1 matched perperty and render tables 1`] = `
Array [
  <div
    className="earthquake-wrapper"
    id="earthquake-detail"
  >
    <h3>
      title
    </h3>
    <table>
      <tbody>
        <tr>
          <td>
            Title
          </td>
          <td>
            title
          </td>
        </tr>
        <tr>
          <td>
            Magnitude
          </td>
          <td>
            mag
          </td>
        </tr>
        <tr>
          <td>
            Time
          </td>
          <td>
            time
          </td>
        </tr>
        <tr>
          <td>
            Status
          </td>
          <td>
            status
          </td>
        </tr>
        <tr>
          <td>
            Tsunami
          </td>
          <td>
            tsunami
          </td>
        </tr>
        <tr>
          <td>
            Type
          </td>
          <td>
            type
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
]
`;

exports[`MyComponent should render my component 1`] = `
Array [
  <div
    className="earthquake-wrapper"
    id="earthquake-detail"
  >
    <h3>
      title
    </h3>
    <table>
      <tbody>
        <tr>
          <td>
            Title
          </td>
          <td>
            title
          </td>
        </tr>
        <tr>
          <td>
            Magnitude
          </td>
          <td>
            mag
          </td>
        </tr>
        <tr>
          <td>
            Time
          </td>
          <td>
            time
          </td>
        </tr>
        <tr>
          <td>
            Status
          </td>
          <td>
            status
          </td>
        </tr>
        <tr>
          <td>
            Tsunami
          </td>
          <td>
            tsunami
          </td>
        </tr>
        <tr>
          <td>
            Type
          </td>
          <td>
            type
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
]
`;
